plugins/s3-viewer-backend/src/service/plugin.ts

```diff
@@ -80,12 +72,14 @@ export const s3ViewerPlugin = createBackendPlugin({
         if (s3BucketStatsProvider) {
           builder = builder.setBucketStatsProvider(s3BucketStatsProvider);
         }
-        if (deps.config.getOptionalBoolean('s3.permissionMiddleware')) {
-          builder = await builder.useMiddleware(s3Middleware);
-        }
 
         const { router } = await builder.build();
         deps.httpRouter.use(router);
+
+        deps.httpRouter.addAuthPolicy({
```

Comment:
I think a short comment here would be good, something like "Allow access with Backstage user-cookie as requests happen client-side from `img` elements which can't set the token via fetchApi".  (I think that's an accurate description, please adjust it if it's not.)

Reply:
Added a tiny comment as suggested